app/routes/questionnaire.py

```diff
@@ -179,22 +182,41 @@ def get_section(schema, questionnaire_store, section_id, list_item_id=None):
         questionnaire_store.metadata,
     )
 
+    if section_id not in router.enabled_section_ids:
+        return redirect(url_for(".get_questionnaire"))
+
+    location = Location(
+        section_id=section_id,
+        list_name=schema.get_repeating_list_for_section(section_id),
+        list_item_id=list_item_id,
+    )
+    section_summary = SectionSummary(schema, questionnaire_store, location, router)
+
+    if request.method == "POST":
+        return redirect(section_summary.get_next_location_url())
+
+    if section_summary.can_access_section_summary():
+        return _render_page(
+            block_type="SectionSummary",
+            context=section_summary.context(),
+            current_location=section_summary.current_location,
+            previous_location_url=section_summary.get_previous_location_url(),
+            schema=schema,
+            page_title=section_summary.page_title,
+        )
+
     if not schema.is_hub_enabled():
         redirect_location = router.get_first_incomplete_location_in_survey()
         return redirect(redirect_location.url())
 
-    if section_id not in router.enabled_section_ids:
-        return redirect(url_for(".get_questionnaire"))
-
-    routing_path = router.routing_path(section_id=section_id, list_item_id=list_item_id)
     section_status = questionnaire_store.progress_store.get_section_status(
         section_id=section_id, list_item_id=list_item_id
     )
 
+    routing_path = router.routing_path(section_id=section_id, list_item_id=list_item_id)
+
     if section_status == CompletionStatus.COMPLETED:
-        return redirect(
-            router.get_section_return_location_when_section_complete(routing_path).url()
-        )
+        return redirect(router.get_first_location_in_section(routing_path).url())
```

Comment:
I think quite a bit of this logic can be simplified by dealing with the `GET` first (as we do in the other methods). I think the logic would be something like:

```
if GET:
    if section not complete:
        redirect to first incomplete location
    if summary in section:
        render summary
    else:
        redirect to first location
```